perceiver_pytorch/mixed_latents.py

```diff
@@ -97,29 +89,25 @@ def __init__(
                 )
             )
 
-        self.to_logits = nn.Sequential(
-            nn.LayerNorm(latent_dim), nn.Linear(latent_dim, num_classes)
-        )
+        self.to_logits = nn.Sequential(nn.LayerNorm(latent_dim), nn.Linear(latent_dim, num_classes))
 
     def forward(self, data, mask=None):
         b, *axis, _, device = *data.shape, data.device
-        assert (
-            len(axis) == self.input_axis
-        ), "input data must have the right number of axis"
+        assert len(axis) == self.input_axis, "input data must have the right number of axis"
 
         # calculate fourier encoded positions in the range of [-1, 1], for all axis
 
         axis_pos = list(
             map(
-                lambda size: torch.linspace(
-                    -1.0, 1.0, steps=size, device=device
-                ),
+                lambda size: torch.linspace(-1.0, 1.0, steps=size, device=device),
                 axis,
             )
         )
         pos = torch.stack(torch.meshgrid(*axis_pos), dim=-1)
         enc_pos = fourier_encode(
-            pos, self.max_freq, self.num_freq_bands, base=self.freq_base
+            pos,
```

Comment:
In general, I'd recommend including the parameter names when calling functions with more than one parameter.  It is a bit excessively verbose!  But it protects us against nasty bugs that might occur if we change the order of the parameters in the function signature (which is unlikely... but not impossible!).

So, here, it would be:

```python
enc_pos = fourier_encode(
    x=pos,
    max_freq=self.max_freq,
    num_bands=self.num_freq_bands
)
```

Which, yeah, doesn't look amazing...  hmm... feel free to not do this!  I'm not entirely sure what's best!